### PR TITLE
Adds a sleeve of tiling colors for several holidays.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
Inspired by #79108. This concerns Christmas, many national holidays and a few gimmicky ones where it'd make sense.

Oh, yeah, I've also added "Sacrebleu" to the list of possible station prefixes for the Bastille day, since "Merde" is already there.

<details>
  <summary>some screenshots (got tired of restarting the server over and over halfway through)</summary>

### St. Patrick (Ireland) : 
Booze Complex Fifty-Four 2023-03-17 090724

### Bastille Day (France) : 
Merde Tomb Seventeen 2023-07-14 084422

### Waitangi Day (Union Jack colors, New Zealand) :
**No, it isn't the french flag again, it's the tram tiles that always display the pattern in vertical stripes...**
Southern Cross Outpost 39 2023-02-06 012622

### Christmas  (Before I realized it looks like the italian flag so I've removed the white) : 
Festive Space-hulk XLIII 2023-12-25 013720
</details>

## Why It's Good For The Game
Implemeting a smidge of festivity and/or celebration to several holidays. Open to suggestions and thoughts.

## Changelog

:cl:
image: Several holidays now have themed floor and tram tiling.
/:cl:

### CODE REVIEW

**Comment by Ghommie** [comment]
Welp,  I'll be on my way to run a local server fourteen times circa in order to provide the necessary screenshots for it.

**Comment by Fikou** [comment]
christmas is fine but i dont want every other day to have the station be covered in random colors

**Comment by Ghommie** [comment]
You make a fine point, I'll roll back several of them.

File: code/modules/holiday/holidays.dm [hunk 1]
```diff
@@ -414,9 +443,15 @@
...
+	holiday_colors = list(
+		COLOR_FRENCH_BLUE,
+		COLOR_WHITE,
+		COLOR_FRENCH_RED
+	)
+	holiday_pattern = PATTERN_VERTICAL_STRIPE
 
 /datum/holiday/france/getStationPrefix()
-	return pick("Francais","Fromage", "Zut", "Merde")
+	return pick("Francais","Fromage", "Zut", "Merde", "Sacrebleu")
```
Comment (jlsnow301) [inline code/modules/holiday/holidays.dm:454]:
```suggestion
	return pick("Francais", "Fromage", "Zut", "Merde", "Sacrebleu")
```
unbelievable

File: code/datums/station_traits/_station_trait.dm [hunk 2]
```diff
@@ -65,9 +65,9 @@
 	qdel(src)
 
 ///Called by decals if they can be colored, to see if we got some cool colors for them. Only takes the first station trait
-/proc/request_station_colors(atom/thing_to_color, pattern = PATTERN_DEFAULT)
+/proc/request_station_colors(atom/thing_to_color, pattern)
 	for(var/datum/station_trait/trait in SSstation.station_traits)
-		var/decal_color = trait.get_decal_color(thing_to_color, pattern)
+		var/decal_color = trait.get_decal_color(thing_to_color, pattern || PATTERN_DEFAULT)
```
Comment (jlsnow301) [inline code/datums/station_traits/_station_trait.dm:70]: is this not doing the same thing?

Comment (Ghommie) [reply code/datums/station_traits/_station_trait.dm:70]: Yes, but it also ensures that the `pattern arg` for the `get_decal_color` proccall defaults to `PATTERN_DEFAULT` when null. See L53 of the decal file.

**Pull request merged**